brayns/io/ColorMapLoader.cpp

```diff
@@ -0,0 +1,121 @@
+/* Copyright (c) 2015-2016, EPFL/Blue Brain Project
+ * All rights reserved. Do not distribute without permission.
+ * Responsible Author: Cyrille Favreau <cyrille.favreau@epfl.ch>
+ *
+ * This file is part of Brayns <https://github.com/BlueBrain/Brayns>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include <brayns/common/log.h>
+#include <brayns/common/material/Texture2D.h>
+#include <brayns/common/scene/Scene.h>
+
+#include <fstream>
+
+#include "ColorMapLoader.h"
+
+namespace brayns
+{
+
+const uint DEFAULT_ALPHA = 255;
+
+ColorMapLoader::ColorMapLoader()
+{
+}
+
+const Vector4uis& ColorMapLoader::getValues() const
+{
+    return _values;
+}
+
+void ColorMapLoader::setValues( const Vector4uis& values )
+{
+    _values.clear();
+    for( auto value: values )
+        _values.push_back( value );
+}
+
+bool ColorMapLoader::loadFromFile( const std::string& filename )
+{
+    BRAYNS_INFO << "Loading color map from " << filename << std::endl;
+    std::ifstream file( filename, std::ios::in );
+    if( !file.good( ))
+    {
+        BRAYNS_ERROR << "Could not open file " << filename << std::endl;
+        return false;
+    }
+
+    bool validParsing = true;
+    std::string line;
+
+    while( validParsing && std::getline( file, line ))
+    {
+        std::vector< uint > lineData;
+        std::stringstream lineStream(line);
+
+        size_t value;
+        while( lineStream >> value )
+            lineData.push_back(value);
+
+        switch( lineData.size() )
+        {
+        case 3:
+            _values.push_back( Vector4ui(
+                lineData[0], lineData[1], lineData[2], DEFAULT_ALPHA ));
+            break;
+        case 4:
+            _values.push_back( Vector4ui(
+                lineData[0], lineData[1], lineData[2], lineData[3] ));
+            break;
+        default:
+            BRAYNS_ERROR << "Invalid line: " << line << std::endl;
+            validParsing = false;
+            break;
+        }
+    }
+
+    file.close();
+    return validParsing;
+}
```

Comment:
My undestanding of the common part is in the definition of the colormap, and how to generate it, not in the way it is implemented in the application itself. I am of course willing to define a common way to generate color maps, if of course this makes sense considering the features provided by the various renderers. The current implementation does not define the color map, it only takes a set of values and makes them available to the simulation renderer. In my opinion, the common part resides outside of the application, for example in the form of a python library that we could you in a notebook (just an idea). I understand you already have some code there that I will be happy to reuse.
Regarding the last question, there is indeed an urgent request for media production that requires this feature.